core/modules/image/image.module

```diff
@@ -781,8 +783,14 @@ function image_style_deliver($style, $scheme) {
   if ($success) {
     image_style_remove_allowed_uri($derivative_uri);
 
-    $image = image_load($derivative_uri);
-    file_transfer($image->source, array('Content-Type' => $image->info['mime_type'], 'Content-Length' => $image->info['file_size']));
+    $extension = pathinfo($derivative_uri, PATHINFO_EXTENSION);
+    if($extension == 'webp') {
+      file_transfer($derivative_uri, array('Content-Type' => 'image/webp', 'Content-Length' => filesize($derivative_uri)));
+    }
+    else {
+      $image = image_load($derivative_uri);
+      file_transfer($image->source, array('Content-Type' => $image->info['mime_type'], 'Content-Length' => $image->info['file_size']));
+    }
```

Comment:
This is a superficial opinion, early warning, as I know nothing about image manipulation generally. But it seems we are trying to overcome the fact that certain image types are hardcoded, by hardcoding `webp`. Could we make this  block more generic if as it seems the `image_load()` call isnt absolutely necessary?
```
    $extension = pathinfo($derivative_uri, PATHINFO_EXTENSION);
    $content_type = some_function_to_get_image_type($derivative_uri);
    file_transfer($derivative_uri, array('Content-Type' =>  $content_type, 'Content-Length' => filesize($derivative_uri)));
```